src/analysis/seminorm.lean

```diff
@@ -313,8 +313,37 @@ calc p 0 = p ((0 : 𝕜) • 0) : by rw zero_smul
 end smul_with_zero
 end add_monoid
 
+section module
+variables [add_comm_group E] [add_comm_group F] [add_comm_group G]
+variables [module 𝕜 E] [module 𝕜 F] [module 𝕜 G]
+
+/-- Composition of a seminorm with a linear map is a seminorm. -/
```

Comment:
Another good pair of lemmas to have would be that `zero_comp : comp 0 f = 0` and `comp_zero : comp p 0 = 0` .

Once the `has_add` PR goes though, you can probably do the same for `add_comp` and `comp_add`.

Reply:
Alright, the `add` PR is merged - mind adding these too?

Reply:
I have a weird issue with adding `comp_smul`: for using `c • f` for ` (f : E →ₗ[𝕜] F) (c : 𝕜)` I need `[normed_field 𝕜]`, but
for `seminorm.comp` I need `[semi_normed_ring 𝕜]` and these two instances don't seem to work together.
````
variables [semi_normed_ring 𝕜] [normed_field 𝕜] [add_comm_group E] [module 𝕜 E]
variables (f : E →ₗ[𝕜] E) (c : 𝕜) (p : seminorm 𝕜 E)

#check c • f
#check seminorm.comp p
````
this fails for first check if `[normed_field 𝕜]` is not present and for the second if `[normed_field 𝕜]` is added.

Reply:
`[normed_field 𝕜]` most definitely imply `[semi_normed_ring 𝕜]`. Don't have them both 😉

Reply:
yes, but if I only have `[normed_field 𝕜]`, then `seminorm.comp p` complains.

Reply:
I found the error..

Reply:
```
variables [normed_field 𝕜] [add_comm_group E] [module 𝕜 E]
variables (f : E →ₗ[𝕜] E) (c : 𝕜) (p : seminorm 𝕜 E)

#check c • f
#check seminorm.comp p
```
works for me. Are you sure you didn't try that in a section which already had instances on `𝕜`? I structured the file by instances on `𝕜` first, so this should go under `section normed_field`.

Reply:
yes, I was stupid and assumed that the `semi_normed_ring` section would not span over to the seminorm ball part.